lib/gmail-adapter.js

```diff
@@ -43,15 +58,19 @@ var Read = Juttle.proc.base.extend({
 
         // XXX/mstemm not sure how the time zone is set. It's definitely not in UTC.
 
-        if (self.options.from) {
-            search += " after:" + JuttleMoment.format(self.options.from, "YYYY/MM/DD", "US/Pacific");
+        if (from) {
+            search += " after:" + JuttleMoment.format(from, "YYYY/MM/DD", "US/Pacific");
         }
 
-        if (self.options.to) {
-            search += " before:" + JuttleMoment.format(self.options.to, "YYYY/MM/DD", "US/Pacific");
+        if (to) {
+            // Add 1 day to to and then quantize to a day. This
+            // results in the next day, such that the before: date is
+            // always in the future.
+            search += " before:" + JuttleMoment.format(to.add(JuttleMoment.duration(1, "d")), "YYYY/MM/DD", "US/Pacific");
```

Comment:
This was the bug I alluded to: if -to was at some point during the day of, say, 2016-01-04, the date was being quantized down to 2016-01-04 and the gmail search was doing "before:2016-01-04". However, that would return messages with dates before but not including 2016-01-04, so any messages during the day of 2016-01-04 were being missed. So add 24 hours and then quantize, so you always catch everything of the current day.